service/models.py

```diff
@@ -53,8 +53,47 @@ class PromoType(Enum):
 
 
 class Product(db.Model):
+    """
+    Class that represents a Product, which just has an ID and no further information.
+
+    This version uses a relational database for persistence which is hidden
+    from us by SQLAlchemy's object relational mappings (ORM)
+    """
+
     id = db.Column(db.Integer, primary_key=True)
 
+    def create(self):
+        """
+        Creates a Product in the database
+        """
+        logger.info("Creating Product with id: %s", self.id)
+        db.session.add(self)
+        db.session.commit()
+
+
+    @classmethod
+    def all(cls):
+        """ Returns all of the Products in the database """
+        logger.info("Processing all Products")
+        return cls.query.all()
+
+    def serialize(self):
+        """ Serializes a Product into a dictionary """
+        return {"id": self.id}
+
+    def deserialize(self, data):
+        """
+        Deserializes a Product from a dictionary
+        """
+        try:
+            self.id = data["id"]
+        except KeyError as error:
+            raise DataValidationError("Invalid promotion: missing id field")
```

Comment:
This might be a typo?
```suggestion
            raise DataValidationError("Invalid product: missing id field")
```